### ISSUE TITLE
Wrong logic of function `addRelationMorph` in '/strapi-hook-mongoose/lib/relations.js`

### ISSUE THREAD

**Issue by ramlez** [strapi/strapi#4382]
Hello guys,
**Describe the bug**
I'm trying to use polymorphic relations in Strapi however every time when I save a document  strapi overrides the array with relations instead of adding new relations to it.

Relation details: 
**Image.settings.json**
```js
{
   // ...
 "attributes": {
    "related": {
      "collection": "*",
      "filter": "field",
      "configurable": false
    },
}
```

**Test.settings.json**
```js
{
   // ...
  "attributes": {
    "type": {
      "type": "string",
      "required": true,
      "unique": true,
      "configurable": true
    }
  }
}
```

**Steps to reproduce the behavior**
1. Create a single **Image** document
2. Create 2 different documents based on **Test** model - both should refer to the same Image
3. The `related` array on Image document (in MongoDb) will contain only 1 entry 
```js
    "related" : [ 
        {
            "_id" : ObjectId("5db6fe8e3972213740a54857"),
            "ref" : ObjectId("5db6f769e69d90323e808ba9"),
            "kind" : "MypluginTest",
            "field" : "FirstImage"
        }
    ],
```
**Expected behavior**
Related array shouldn't be overridden.

**Code**
In my humble opinion the code in the function `addRelationMorph` in `/strapi-hook-mongoose/lib/relations.js` is wrong as the `const value = []` will always give an empty array. It seems that value should be populated based on the existing value from the database.  Someting like:
```js
const value = entry[params.alias] || []; // not tested yet
```
In the current solution checking if some relation already exists in array will return `false` every time. Here is the full code of that method: 
```js
addRelationMorph: async function(params) {
    let entry = await this.findOne({
      [this.primaryKey]: getValuePrimaryKey(params, this.primaryKey),
    });

    if (entry) {
      entry = entry.toJSON();
    }

    const value = [];

    // Retrieve association.
    const association = this.associations.find(
      association => association.alias === params.alias
    );

    if (!association) {
      throw Error(
        `Impossible to create relationship with ${params.ref} (${params.refId})`
      );
    }

    // Resolve if the association is already existing.
    const isExisting = value.find(obj => {
      if (
        obj.kind === params.ref &&
        obj.ref.toString() === params.refId.toString() &&
        obj.field === params.field
      ) {
        return true;
      }

      return false;
    });

    // Avoid duplicate.
    if (isExisting) {
      return Promise.resolve();
    }

    // Push new relation to the association array.
    value.push({
      ref: params.ref,
      refId: params.refId,
      kind: params.ref,
      field: params.field,
    });

    entry[params.alias] = value;

    return module.exports.update.call(this, {
      id: params.id,
      values: entry,
    });
  },
```

**System**
- Node.js version: v10.16.3
- NPM version: 6.9.0
- Strapi version: 3.0.0-beta.17.1
- Database: MongoDb
- Operating system: Ubuntu 18

I hope I'm right and I haven't missed anything important. ;) 

Regards,

**Comment by lauriejim** [comment]
Thank you @ramlez for reporting this.
I ping @alexandrebodin.

**Comment by derrickmehaffy** [comment]
@ramlez you win my award for most detailed bug report I've seen in a while thank you.

Side note can you also test with the latest Strapi version, I don't think there was any changes to this but would be good to know. `Strapi version: 3.0.0-beta.17.1` Current latest is beta.17.4

**Comment by alexandrebodin** [comment]
Hi @ramlez, I think you are right  :) 

There is indeed an issue in this fonction. I won't be able to tackle it right away but If you feel adventurous you can try and make a PR to fix this ;)

**Comment by alexandrebodin** [comment]
Hi @ramlez, This issue has been fixed in versions >  3.0.0-beta.18 of strapi.

I'm closing this issue. Let us know if you still have issues concerning this :)
